internal/services/maintenance/maintenance_assignment_dedicated_host_resource.go

```diff
@@ -152,8 +158,15 @@ func resourceArmMaintenanceAssignmentDedicatedHostRead(d *pluginsdk.ResourceData
 	}
 	d.Set("dedicated_host_id", dedicatedHostId)
 
-	if props := assignment.ConfigurationAssignmentProperties; props != nil {
-		d.Set("maintenance_configuration_id", props.MaintenanceConfigurationID)
+	if props := assignment.Properties; props != nil {
+		if props.MaintenanceConfigurationId != nil {
+			maintenanceConfigurationId := ""
+			if props.MaintenanceConfigurationId != nil {
+				maintenanceConfigurationId = *props.MaintenanceConfigurationId
+			}
+			d.Set("maintenance_configuration_id", maintenanceConfigurationId)
+		}
+
```

Comment:
we need to ensure that a value is always set into the state for this, plus we should normalize it on the way back which'd solve the casing difference

```suggestion
		maintenanceConfigurationId := ""
		if props.MaintenanceConfigurationId != nil {
			parsedId, err = maintenanceconfigurations.ParseMaintenanceConfigurationIDInsensitively(*props.MaintenanceConfigurationId)
			if err != nil {
				return fmt.Errorf("parsing %q: %+v", *props.MaintenanceConfigurationId, err)
			}
			maintenanceConfigurationId = parsedId.ID()
		}
		d.Set("maintenance_configuration_id", maintenanceConfigurationId)

```